### DISCUSSION TITLE
Naming of joints for single group

### DISCUSSION THREAD

**Discussion by jimmy-mcelwain** [Yaskawa-Global/motoros2#331] [General]
We mentioned this in our meeting last week, but I just want to bring it up again because I'm working on the MoveIt2 package instructions for Motoman robots. Setting everything in the urdf/xacro files to `group_1/joint_1`, `group_1/joint_2`... gets a little tedious and is an easy place to make a mistake. We talked about changing the default for single group to be `joint_1`, `joint_2`... and omitting the group prefix. Is that something that we want to do? I don't know all of the drawbacks, but it would make MoveIt easier to set up. If we do that, would multi-group still be `group_1/joint_1`, `group_1/joint_2`, ... ,`group_2/joint_1`, `group_2/joint_2`...?

**Comment by ted-miller** [comment]
I personally had a real bad time trying to rename everything properly in the xacro file. I think simpler == gooder.

We've already got a second publisher for `ctrl_groups/r1/joint_states` if someone wants to be more explicit.

So I'm in favor of this idea.

> If we do that, would multi-group still be group_1/joint_1, group_1/joint_2, ... ,group_2/joint_1, group_2/joint_2

I think we'd have to.

**Comment by gavanderhoorn** [comment]
> We mentioned this in our meeting last week, but I just want to bring it up again because I'm working on the MoveIt2 package instructions for Motoman robots. 

while I believe it would be good to change joint naming for single-group systems, that's not implemented yet. Updating your current instructions for functionality that's not in MotoROS2 -- especially not in released versions -- might not be something you'd want to do.

I'd suggest documenting how things currently work, and updating at a later point when we've changed them.

> Setting everything in the urdf/xacro files to `group_1/joint_1`, `group_1/joint_2`... gets a little tedious and is an easy place to make a mistake. We talked about changing the default for single group to be `joint_1`, `joint_2`... and omitting the group prefix. Is that something that we want to do? I don't know all of the drawbacks, but it would make MoveIt easier to set up.

For MoveIt, joint names would only come up in the controller configuration if I'm not mistaken.

Everything else uses `link`s, which are not affected by all of this.

> If we do that, would multi-group still be `group_1/joint_1`, `group_1/joint_2`, ... ,`group_2/joint_1`, `group_2/joint_2`...?

yes, `joint` names must always be unique, so they must have some prefix. At that point using generic group identifiers would be preferred.
